### PR TITLE
ci: add python 3.13 to test matrix

### PR DESCRIPTION
**Pull request by venikkin**

### CODE REVIEW

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -13,11 +13,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12" ]
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
```
Comment (ali-ince) [inline .github/workflows/python-ci.yml:16]: `3.8` is also EOL'd earlier this month, should we remove that too?

https://devguide.python.org/versions/

File: .github/workflows/python-ci.yml [hunk 2]
```diff
@@ -13,11 +13,14 @@ jobs:
...
         neo4j-version: [ "4.4", "5" ]
         spark-version:
           - { short: "3", ext: "3.4.3", scala: "2.12" }
           - { short: "3", ext: "3.5.2", scala: "2.12" }
+        exclude:
+          - python-version: "3.13"
+          - spark-version: { short: "3", ext: "3.4.3", scala: "2.12" }
```
Comment (ali-ince) [inline .github/workflows/python-ci.yml]: Does this not exclude all python 3.13 builds?

Should it look like this instead?

```suggestion
        exclude:
          - python-version: "3.13"
            spark-version: { short: "3", ext: "3.4.3", scala: "2.12" }
```

Comment (fbiville) [reply .github/workflows/python-ci.yml]: argh 🤦 let me fix this

File: .github/workflows/python-ci.yml (hunk 1 above)
Comment (fbiville) [reply .github/workflows/python-ci.yml:16]: follow-up PR OK for this?

Comment (fbiville) [reply .github/workflows/python-ci.yml:16]: https://github.com/neo4j/neo4j-spark-connector/pull/678

**Review by ali-ince — APPROVED** [review]

**Pull request merged**